### PR TITLE
Implement simple fromNow() API based on timeago.js

### PR DESCRIPTION
**Pull request by aalises**
Implementation of the fromNow() function of moment based on timeago.js . The API is as follows:

```
dayjs().fromNow(dayjs)
```
And returns a string with the time passed between the two dates with the following format:
```
Just now , in X hours/years/months , X hours/years/months ago ...
```
For the sake of simplicity, the time is adjusted to the highest time measurement instead of being unit defined, e.g 60 days will be 2 months , 94670856 seconds will be 2 years, and 76 seconds will be 1 minute , and so on.

Tests are also implemented and passing, which compare the function output with timeago() in various scenarios.

TO DO: Maybe adding some template to change the locale of the messages (change X years ago for X years, or just X...) or to be able to customize them as moment, but to be fair I don't see it as necessary, just adding the chinese version should be it, but it could certainly be implemented. Adding a few lines of code, the version with a specific unit specified (I want it defined in weeks), can also be put, but that would be maybe complicating it a lot?

### CODE REVIEW

**Comment by iamkun** [comment]
Perfect thanks. see my comment #50

File: src/index.js [hunk 1]
```diff
@@ -282,6 +282,22 @@ class Dayjs {
...
+    const P = [{ l: C.S, v: C.MILLISECONDS_A_SECOND }, { l: C.M, v: C.MILLISECONDS_A_SECOND * 60 },
+      { l: C.H, v: C.MILLISECONDS_A_HOUR }, { l: C.D, v: C.MILLISECONDS_A_DAY },
+      { l: C.W, v: C.MILLISECONDS_A_WEEK }, { l: C.M, v: C.MILLISECONDS_A_WEEK * 4 },
+      { l: C.Q, v: C.MILLISECONDS_A_WEEK * 12 }, { l: C.Y, v: C.MILLISECONDS_A_WEEK * 4 * 12 }]
+
+    const result = input.diff(this, `${C.MS}s`)
+    const resabs = Math.abs(result)
+    let out = ''
+    for (let i = 0; i < P.length; i += 1) {
+      if (resabs >= P[i].v) out = `${Utils.absFloor(resabs / P[i].v)} ${P[i].l}${(resabs / P[i].v !== 1) ? 's' : ''}`
```
[the remark this answers is not in the document]
Comment (aalises) [reply src/index.js]: Thank you very much @Imperat ! it is indeed cleaner and faster. Just pushed the change (some merge conflicts have to be resolved though :+1: )

[the remark this answers is not in the document]
Comment (Imperat) [reply src/index.js]: Thank you, @aalises!

**Comment by iamkun** [comment]
@aalises Thanks your lovely work. It's great. 

Some questions:
- why compare this with timeago.js  instead of moment.fromNow in test case?
- don't understand here
  > spanish local  maybe using getters 
- Thanks for the bug fix. 
   > Also fixed the add method with the locales

Something I'm thinking.
I think we might better add fromNow as a plugin instead of adding to the core code. We may have to add a lot more functions to meets others need such as timezone, UTC support, calendar time in the future. And if we add every thing to our core code, we can't control the total size. (Another moment in size may be). That's why we have a plugin system now.

And we will treat these plugins (mentioned above ) as Day.js extensions with official support rather than user's customized plugin. And added these to our main repo and released with core and locale to NPM.

And if you agree with me, another problem comes. Should we add locales used by plugins to our main locale file? If so, the bundle size will increase. If not, how could we ship these plugins' locale?

P.S. join us at #123 about locale && plugin

**Comment by aalises** [comment]
@iamkun of course! making a plugin out of it would be great. Timeago.js was used for the test cases for its simplicity, but we can use moment as well! and the getter thing was to simplify the locale by reusing object properties to be part (as a string) of another one, but nvm.

**Comment by iamkun** [comment]
I tend to use moment.js. And any simple example code of 'getter'? Still can't get it, sad.

**Comment by aalises** [comment]
@iamkun implemented fromNow as a plugin! Still have to figure out what to do with the locales as the plugin sort of extend them (adding, the future, past and now fields and the plural forms), should we put this fields separately inside the plugin or just add them to the locales?

**Comment by iamkun** [comment]
Nice plugin! 
I think, the en (english) locale should comes with plugin, while others should be in separate locales respectively.  How do you think?
see [](https://github.com/xx45/dayjs/blob/master/src/plugin/advancedFormat.js#L7)  and [](https://github.com/xx45/dayjs/pull/141/files#diff-0261a68f99591aea1daf321474b982acR5)
But would you mind just wait a moment. There's a little problem with our locale system and the bundle packager. I will fix it in #141. And there might be some code or folder structure change.

**Comment by aalises** [comment]
Perfect! let me know when it's fixed :+1:  So you suggest putting them on plugin/locale?

**Comment by iamkun** [comment]
Sure. I mean just put them on `locale/es` but add en locale to the plugin itself, cause its our default locale. You could check the link in my reply above. But still, I'm not sure if it is the best solution.

File: test/display.test.js [hunk 2]
```diff
@@ -155,44 +155,44 @@ describe('Difference', () => {
       expect(dayjsA.diff(dayjsC, unit, true)).toBe(momentA.diff(momentC, unit, true))
     })
   })
-})
```
Comment (iamkun) [inline test/display.test.js]: why this removal? and why the formatting blew?

File: package.json [hunk 3]
```diff
@@ -6,7 +6,7 @@
   "types": "index.d.ts",
   "scripts": {
     "test": "jest",
-    "lint": "./node_modules/.bin/eslint src/* test/* build/*",
+    "lint": "./node_modules/.bin/eslint src/* test/* build/* --fix",
```
Comment (iamkun) [inline package.json]: no `--fix`, this could cause a CI pass(auto fix) but not qualified code.

**Comment by aalises** [comment]
@iamkun yes! moving it into the plugin would keep the constant file leaner, but any other locale needs to have the plural forms implemented.

And I used timeago.js instead of moment because moment is more complex and wouldn't pass the checks (timeago only has X time ago, in X time, and now), while moment has a lot more semantics. I preferred to keep it simple.

**Comment by iamkun** [comment]
All right, just timeago.js at the moment.

Plus, what I mean is to move en locale to plugin only, and leave the reset to their local files.

That is to say, all plugin comes with a built in en locale, the same as core module. But if you want to use other locales, you have to import them ahead.

**Comment by iamkun** [comment]
Also, consider moment.js's local file:

```javascript
relativeTime : {
            future : 'en %s',
            past : 'hace %s',
            s : 'unos segundos',
            ss : '%d segundos',
            m : 'un minuto',
            mm : '%d minutos',
            h : 'una hora',
            hh : '%d horas',
            d : 'un día',
            dd : '%d días',
            M : 'un mes',
            MM : '%d meses',
            y : 'un año',
            yy : '%d años'
        },
```

locale comes with a namespace `relativeTime` looks easy to understand.

And is `un minuto`, `1 minuto` the same? Don't know Spanish 😬 .

**Comment by iamkun** [comment]
plugin(option, Dayjs, dayjs)

so d for dayjs function.

you could move your code above proto... jist like advancedFormat.

And a namespace is needed so that you could just do 

   d.en.relativeTime = {} 

instead of merge the object.

replied on my phone, sorry for the code format.

Plus, sorry for the lack of docs, I'll finish it tomorrow.

**Comment by aalises** [comment]
@iamkun great! will make those changes sometime tomorrow when I can find some spare time :)

**Comment by aalises** [comment]
@iamkun I pushed the changes yesterday night. With that, everything should be done by now I think. Now the locales implement relativeTime for the plugin. WDYT?

**Comment by iamkun** [comment]
looks great. BTW, what's the reason for changing or formatting ` test/display.test.js` file ?

**Comment by iamkun** [comment]
@aalises I think it's not because of the lint auto-fixer. Seems you've deleted one line of code L#158.

**Comment by aalises** [comment]
@iamkun Yes, when I first coded the fromNow tests they were on the display.test files under the Difference describe block. When moving them to a sepparate file y accidentally deleted the closing bracket it seems, but did not crash because it was still opened at the bottom. Now the block is correctly scoped

**Comment by iamkun** [comment]
@aalises Got it. Looks good to me. Will merge into next release. THX 😀

**Comment by aalises** [comment]
@iamkun perfect thanks! we should put on the documentation some template for the locales as now the relativeTime should be part of it in order for the plugin to work.

**Comment by iamkun** [comment]
@aalises I am doing the docs. tough work 😬

**Comment by iamkun** [comment]
Will release in 1.6.3

**Comment by chao** [comment]
Great!

**Pull request merged**